packages/go/analysis/ad/esc9.go

```diff
@@ -0,0 +1,168 @@
+					} else if subjRequireDns || subjRequireDomainDns {
+						//If either of these properties is true, we need to remove all these users from our victims list
+						victimBitmap.Xor(cardinality.NodeSetToDuplex(userNodes))
+					}
+				}
+
+				if attackers, err := ops.FetchStartNodes(tx.Relationships().Filterf(func() graph.Criteria {
+					return query.And(
+						query.KindIn(query.Start(), ad.Group, ad.User, ad.Computer),
+						query.KindIn(query.Relationship(), ad.GenericAll, ad.GenericWrite, ad.Owns, ad.WriteOwner, ad.WriteDACL),
+						query.InIDs(query.EndID(), cardinality.DuplexToGraphIDs(victimBitmap)...),
+					)
+				})); err != nil {
+					return err
+				} else {
+					results.Or(cardinality.NodeSetToDuplex(attackers))
+				}
+			}
+		}
+
+		results.Each(func(value uint32) (bool, error) {
+			if !channels.Submit(ctx, outC, analysis.CreatePostRelationshipJob{
+				FromID: graph.ID(value),
+				ToID:   domain.ID,
+				Kind:   ad.ADCSESC9a,
+			}) {
+				return false, nil
+			} else {
+				return true, nil
+			}
+		})
+
+		return nil
+	}
+}
+
+func expandNodeSliceToBitmapWithoutGroups(nodes []*graph.Node, groupExpansions impact.PathAggregator) cardinality.Duplex[uint32] {
+	var bitmap = cardinality.NewBitmap32()
+	for _, controller := range nodes {
+		if controller.Kinds.ContainsOneOf(ad.Group) {
+			groupExpansions.Cardinality(controller.ID.Uint32()).(cardinality.Duplex[uint32]).Each(func(id uint32) (bool, error) {
+				//Check group expansions against each id, if cardinality is 0 than its not a group
+				if groupExpansions.Cardinality(id).Cardinality() == 0 {
+					bitmap.Add(id)
+				}
+
+				return true, nil
+			})
+		} else {
+			bitmap.Add(controller.ID.Uint32())
+		}
+	}
+
+	return bitmap
+}
+
+func isCertTemplateValidForESC9a(ct *graph.Node) (bool, error) {
+	if reqManagerApproval, err := ct.Properties.Get(ad.RequiresManagerApproval.String()).Bool(); err != nil {
+		return false, err
+	} else if reqManagerApproval {
+		return false, nil
+	} else if authenticationEnabled, err := ct.Properties.Get(ad.AuthenticationEnabled.String()).Bool(); err != nil {
+		return false, err
+	} else if !authenticationEnabled {
+		return false, nil
+	} else if noSecurityExtension, err := ct.Properties.Get(ad.NoSecurityExtension.String()).Bool(); err != nil {
+		return false, err
+	} else if !noSecurityExtension {
+		return false, nil
+	} else if enrolleeSuppliesSubject, err := ct.Properties.Get(ad.EnrolleeSuppliesSubject.String()).Bool(); err != nil {
+		return false, err
+	} else if enrolleeSuppliesSubject {
+		return false, nil
+	} else if schemaVersion, err := ct.Properties.Get(ad.SchemaVersion.String()).Float64(); err != nil {
+		return false, err
+	} else if authorizedSignatures, err := ct.Properties.Get(ad.AuthorizedSignatures.String()).Float64(); err != nil {
+		return false, err
+	} else if schemaVersion > 1 && authorizedSignatures > 0 {
```

Comment:
From the ticket: `SchemaVersion = 1 OR (SchemaVersion > 1 AND AuthorizedSignatures = 0)`
So I guess we should change the check to: `schemaVersion == 1 || (schemaVersion > 1 && authorizedSignatures > 0)`

Reply:
This check works as expected. If the schemaversion is 1, it wont drop into that else statement. If the schemaversion is > 1 and authorizedsignatures is > 0 it'll drop into the else statement and return false

Reply:
right. I read that statement incorrectly